Main.java

```diff
@@ -3,74 +3,121 @@
 import java.io.*;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.sql.*;
 import java.util.StringTokenizer;
 
 public class Main {
 
-    final static String FILE_TYPE_TXT = ".txt";
-    final static String FILE_TYPE_JSON = ".json";
-    final static String REMOVE_SPECIAL_CHAR = "\\p{Punct}";
-    final static String SINGLE_SPACE = "";
+    static final String FILE_TYPE_TXT = ".txt";
+    static final String FILE_TYPE_JSON = ".json";
+    static final String REMOVE_SPECIAL_CHAR = "[^a-zA-Z0-9 -]";
+    static final String SINGLE_SPACE = "";
+    static String searchResult;
 
     public static void main(String[] args) {
 
+        String textFilePath;
+        String searchKeyword;
+        String errorMessage;
         if (args.length != 2) {
-            System.out.println("File path and Search word cannot be empty");
+            errorMessage = "Provide valid filepath and search keyword";
+            System.out.println(errorMessage);
+            Database.databaseOperation(null, null, searchResult = "Failure", errorMessage, 0);
             return;
         }
-        String textFilePath = args[0];
-        String searchKeyword = args[1];
+        textFilePath = args[0];
+        searchKeyword = args[1];
         File file = new File(textFilePath);
         System.out.println("Processing...");
-        if (isValidFile(file)) {
-            wordCount(textFilePath, searchKeyword);
+        try {
+            if (isValidFile(file, textFilePath, searchKeyword)) {
+                if (file.length() != 0) {
+                    wordCount(textFilePath, searchKeyword);
+                } else {
+                    errorMessage = "File does not contain any data";
+                    System.out.println(errorMessage);
+                    Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, 0);
+                }
+            }
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
         }
     }
 
-    private static boolean isValidFile(File file) {
-
+    private static boolean isValidFile(File file, String textFilePath, String searchKeyword) throws ClassNotFoundException {
+        String errorMessage;
         if (!file.exists()) {
-            System.out.println("File does not exist");
+            errorMessage = "File does not exist";
+            System.out.println(errorMessage);
+            Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, 0);
             return false;
         } else if (!(file.getName().endsWith(FILE_TYPE_TXT) || file.getName().endsWith(FILE_TYPE_JSON))) {
-            System.out.println("File format not supported");
+            errorMessage = "File format not supported";
+            System.out.println(errorMessage);
+            Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, 0);

```

Comment:
Passing null is not a recommended approach. Do the check in the method inside the databaseOperation methods and put appropriate messages.

Reply:
Sure. Will do.